packages/deposit_firebase/test/deposit_firebase_test.dart

```diff
@@ -0,0 +1,150 @@
+        'cars',
+        'id',
+        {
+          'brand': 'VW',
+          'model': 'Nivus',
+        },
+      );
+
+      final snapshot = await instance.collection('cars').get();
+      expect(snapshot.docs.length, equals(1));
+      expect(snapshot.docs.first['brand'], equals('VW'));
+      expect(snapshot.docs.first['model'], equals('Nivus'));
+    });
+
+    test('by returns a list of docs', () async {
+      await Future.wait([
+        instance.collection('cars').add({
+          'brand': 'VW',
+          'model': 'Nivus',
+        }),
+        instance.collection('cars').add({
+          'brand': 'VW',
+          'model': 'Virtus',
+        }),
+        instance.collection('cars').add({
+          'brand': 'GM',
+          'model': 'Onix',
+        }),
+      ]);
+
+      final list = await adapter.by('cars', 'brand', 'VW');
+      expect(list.length, equals(2));
+    });
+
+    group('exists', () {
+      group('using the id primary key', () {
+        test('exists returns true when a doc is in the db', () async {
+          final snapshot = await instance.collection('cars').add({
+            'brand': 'VW',
+            'model': 'Nivus',
+          });
+
+          expect(
+            await adapter.exists('cars', 'id', snapshot.id),
+            isTrue,
+          );
+        });
+      });
+
+      group('using a field different than the id', () {
+        test('exists returns true when a doc is in the db', () async {
+          await instance.collection('cars').add({
+            'brand': 'VW',
+            'model': 'Nivus',
+          });
+
+          expect(
+            await adapter.exists('cars', 'brand', 'VW'),
+            isTrue,
+          );
+        });
+      });
+    });
+
+    group('getById', () {
+      test('returns a doc by id', () async {
+        final snapshot = await instance.collection('cars').add({
+          'brand': 'VW',
+          'model': 'Nivus',
+        });
+
+        expect(
+          await adapter.getById('cars', 'id', snapshot.id),
+          isNotNull,
+        );
+      });
+
+      test('throws when a field other than id is used', () async {
+        expect(
+          () => adapter.getById('cars', 'other', '123'),
+          throwsArgumentError,
+        );
+      });
+    });
+
+    test('page throws unsupported', () {
+      expect(
+        () => adapter.page('cars', limit: 1, skip: 0),
+        throwsA(isA<UnsupportedError>()),
+      );
+    });
+
+    test('remove deletes the doc', () async {
+      final data = await adapter.add('cars', 'id', {
+        'brand': 'VW',
+        'model': 'Nivus',
+      });
+
+      await adapter.remove('cars', 'id', data);
+
+      expect(
+        await adapter.exists('cars', 'brand', 'VW'),
+        isFalse,
+      );
+    });
+
+    test('updates a doc', () async {
+      await adapter.add('cars', 'id', {
+        'brand': 'VW',
+        'model': 'Virtus',
+      });
+
+      await adapter.update('cars', 'id', {
+        'brand': 'VW',
```

Comment:
This should need an `Id` that corresponds to the `primaryColumn`, the `update` and `remove` should then validate if the received data has the `primaryColumn`, if not it should never succeeds because we need the primary column to find the correct document.

Reply:
You are right, I changed this.